graders/c/cgrader/cgrader.py

```diff
@@ -76,14 +76,18 @@ def test_compile_file(self, c_file, exec_file, main_file=None,
             add_c_file = []
         elif not isinstance(add_c_file, list):
             add_c_file = [add_c_file]
-        if main_file: # Kept for compatibility reasons, but could be set as an added file
-            add_c_file.append(main_file)
         if add_c_file:
             flags.append('-Wl,--allow-multiple-definition')
 
         if not compiler:
             compiler = self.compiler
 
+        if pkg_config_flags:
+            if isinstance(pkg_config_flags, str):
+                pkg_config_flags = shlex.split(pkg_config_flags)
+            out_flags = self.run_command(['pkg-config', '--cflags', '--libs'] + pkg_config_flags)
+            if out_flags: flags.extend(shlex.split(out_flags))
```

Comment:
Nit for consistency:

```suggestion
            if out_flags:
            		flags.extend(shlex.split(out_flags))
```